### PR TITLE
Add pause to allowed openstack server status

### PR DESCRIPTION
**Pull request by arnaudmorin**

### CODE REVIEW

File: openstack/resource_openstack_compute_instance_v2.go [hunk 1]
```diff
@@ -784,7 +784,7 @@ func resourceComputeInstanceV2Read(_ context.Context, d *schema.ResourceData, me
 	// Set the current power_state
 	currentStatus := strings.ToLower(server.Status)
 	switch currentStatus {
-	case "active", "shutoff", "error", "migrating", "shelved_offloaded", "shelved":
+	case "active", "shutoff", "error", "migrating", "shelved_offloaded", "shelved", "pause":
```
Comment (kayrus) [inline openstack/resource_openstack_compute_instance_v2.go:787]: could you please update the corresponding documentation?

Comment (arnaudmorin) [reply openstack/resource_openstack_compute_instance_v2.go:787]: sure!

**Review by nikParasyr — CHANGES_REQUESTED** [review]
Tests are failing:

```
2024-02-05T12:34:51.5870467Z 2024-02-05T12:34:51.586Z [ERROR] sdk.proto: Response contains error diagnostic: tf_provider_addr=registry.terraform.io/hashicorp/openstack tf_proto_version=5.4 tf_resource_type=openstack_compute_instance_v2 diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="expected power_state to be one of [\"active\" \"shutoff\" \"shelved_offloaded\"], got paused" tf_req_id=a1860b23-a1c4-b558-3d2c-34e87fe510b2 tf_rpc=ValidateResourceTypeConfig diagnostic_attribute="AttributeName(\"power_state\")"
2024-02-05T12:34:51.5913366Z 2024-02-05T12:34:51.590Z [WARN]  sdk.helper_resource: Error running Terraform CLI command: test_name=TestAccComputeV2Instance_initialPaused test_terraform_path=/usr/local/bin/terraform test_step_number=2 test_working_directory=/tmp/plugintest1183558114
2024-02-05T12:34:51.5915707Z   error=
2024-02-05T12:34:51.5916107Z   | exit status 1
2024-02-05T12:34:51.5916507Z   | 
2024-02-05T12:34:51.5917391Z   | Error: expected power_state to be one of ["active" "shutoff" "shelved_offloaded"], got paused
2024-02-05T12:34:51.5918292Z   | 
2024-02-05T12:34:51.5918769Z   |   with openstack_compute_instance_v2.instance_1,
2024-02-05T12:34:51.5919980Z   |   on terraform_plugin_test.tf line 5, in resource "openstack_compute_instance_v2" "instance_1":
2024-02-05T12:34:51.5920955Z   |    5:   power_state = "paused"
2024-02-05T12:34:51.5921407Z   | 
2024-02-05T12:34:51.5921699Z   
2024-02-05T12:34:51.5924144Z 2024-02-05T12:34:51.590Z [DEBUG] sdk.helper_resource: Stopping providers: test_working_directory=/tmp/plugintest1183558114 test_name=TestAccComputeV2Instance_initialPaused test_terraform_path=/usr/local/bin/terraform test_step_number=2
2024-02-05T12:34:51.5928227Z 2024-02-05T12:34:51.591Z [ERROR] sdk.helper_resource: Error running post-test destroy, there may be dangling resources: test_name=TestAccComputeV2Instance_initialPaused test_terraform_path=/usr/local/bin/terraform test_step_number=2 test_working_directory=/tmp/plugintest1183558114
2024-02-05T12:34:51.5930593Z   error=
2024-02-05T12:34:51.5930976Z   | exit status 1
2024-02-05T12:34:51.5931342Z   | 
2024-02-05T12:34:51.5932130Z   | Error: expected power_state to be one of ["active" "shutoff" "shelved_offloaded"], got paused
2024-02-05T12:34:51.5932984Z   | 
2024-02-05T12:34:51.5933454Z   |   with openstack_compute_instance_v2.instance_1,
2024-02-05T12:34:51.5934543Z   |   on terraform_plugin_test.tf line 5, in resource "openstack_compute_instance_v2" "instance_1":
2024-02-05T12:34:51.5935499Z   |    5:   power_state = "paused"
2024-02-05T12:34:51.5935978Z   | 
2024-02-05T12:34:51.5936266Z   
2024-02-05T12:34:51.5937651Z     testing_new.go:85: Error running post-test destroy, there may be dangling resources: exit status 1
2024-02-05T12:34:51.5938535Z         
2024-02-05T12:34:51.5939547Z         Error: expected power_state to be one of ["active" "shutoff" "shelved_offloaded"], got paused
2024-02-05T12:34:51.5940488Z         
2024-02-05T12:34:51.5941127Z           with openstack_compute_instance_v2.instance_1,
2024-02-05T12:34:51.5942502Z           on terraform_plugin_test.tf line 5, in resource "openstack_compute_instance_v2" "instance_1":
2024-02-05T12:34:51.5943551Z            5:   power_state = "paused"
2024-02-05T12:34:51.5944025Z         
2024-02-05T12:34:51.5944596Z --- FAIL: TestAccComputeV2Instance_initialPaused (11.82s)
```

I expect it is because [this](https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/main/openstack/resource_openstack_compute_instance_v2.go#L409) has not been updated

**Comment by kayrus** [comment]
@arnaudmorin can you rebase and fix tests?

**Comment by arnaudmorin** [comment]
Hey, rebase done, will see for tests

**Comment by yomovh** [comment]
@nikParasyr : the test seems to be sucessful. Is it ok to merge the PR ?

**Review by nikParasyr — APPROVED** [review]

**Pull request merged**